evmdis/main.go

```diff
@@ -5,29 +5,96 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-
-	"github.com/Arachnid/evmdis"
+	"log"
+	"strconv"
+	"flag"
+	"github.com/soad003/evmdis"
 )
 
 func main() {
+	
+	withSwarmHash := flag.Bool("swarm", true, "a bool")
+	ctorMode := flag.Bool("ctor", false, "a bool")
+	logging := flag.Bool("log", false, "a bool")
+	//callsMode := flag.Bool("calls", false, "a bool")
+
+	flag.Parse()
+
+	if !*logging {
+		log.SetOutput(ioutil.Discard)
+	}
+
 	hexdata, err := ioutil.ReadAll(os.Stdin)
 	if err != nil {
 		panic(fmt.Sprintf("Could not read from stdin: %v", err))
 	}
 
-	bytecode := make([]byte, hex.DecodedLen(len(hexdata)))
+	bytecodeLength := uint64(hex.DecodedLen(len(hexdata)))
+	bytecode := make([]byte, bytecodeLength)
+
 	hex.Decode(bytecode, hexdata)
 
-	program := evmdis.NewProgram(bytecode)
-	if err := evmdis.PerformReachingAnalysis(program); err != nil {
-		panic(fmt.Sprintf("Error performing reaching analysis: %v", err))
-	}
-	evmdis.PerformReachesAnalysis(program)
-	evmdis.CreateLabels(program)
-	if err := evmdis.BuildExpressions(program); err != nil {
-		panic(fmt.Sprintf("Error building expressions: %v", err))
+	if bytecode[bytecodeLength-1] == 0x29 &&
```

Comment:
Good point. I think a better way in general would be to identify blocks that are never reached, and output them as simple hex strings, rather than trying to print their opcodes.